Add System.DirectoryServices.AccountManagement Tests

tarekgh opened the pull request.

tarekgh commented:
CC @AlexGhiondea @danmosemsft @hughbe @WinCPP

tarekgh commented:
#24590
#24316
#24388
#24533

tarekgh commented:
@tquerec ping!

tquerec commented on src/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs:
```diff
@@ -0,0 +1,510 @@
+
+                using (UserPrincipal cu = UserPrincipal.Current)
+                {
+                    try
```
.Current requires a domain joined client so we might want to mark that in the method dependencies.

tquerec commented on src/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs:
```diff
@@ -0,0 +1,510 @@
+
+        private UserPrincipal CreateUser(PrincipalContext context, string userName, string password, string firstName, string lastName, string displayName)
+        {
+            UserPrincipal user = new UserPrincipal(context, userName, password, true);
```
This will create the object in a well known location ( users container ) which means that concurrent execution of these tests will cause failures.  We should generate a random number during each test execution and append this to the object names that way there won't be any collisions

tarekgh replied:
The only concern with that is if some reason we created a user with random name and then the test crashes (or terminated up normally), we'll not be able to clean it up. do you think this is ok?

tquerec replied:
I thought about this more and I agree that it's more important for us to keep a clean test environment than to support concurrent execution.  We can always add that in the future if required.  Keep the code as is.

tquerec commented on src/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs:
```diff
@@ -0,0 +1,510 @@
+            try
+            {
+                using (PrincipalContext context = DomainContext)
+                using (UserPrincipal p1 = CreateUser(context, "corefxtest1", "netcore#1pass", "corefx1", "test1", "CoreFx Test User 1"))
```
you've got names and credentials copied and hardcoded in a couple places here.  These should be replaced with a single source

tquerec commented on src/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs:
```diff
@@ -0,0 +1,510 @@
+
+namespace System.DirectoryServices.AccountManagement.Tests
+{
+    public class AccountManagementTests
```
The only other test case I would add is PrincipalContext.ValidateCredentials.

tarekgh commented:
@tquerec I have addressed all your feedback. I have avoided generating random users/groups as we discussed. let me know if you have any other comments. Thanks.
